### PR TITLE
document: adapt serializer to filter by org in admin view

### PR DESCRIPTION
**Pull request by AoNoOokami**
* Adapts filter to filter results by org in admin view.
* Removes unused code in post process search serializer.

Co-Authored-by: Alicia Zangger <alicia.zangger@rero.ch>
Co-Authored-by: Johnny Mariéthoz <Johnny.Mariethoz@rero.ch>

## Why are you opening this PR?

This is a partial implementation of task 1377 of US 1208:
https://tree.taiga.io/project/rero21-reroils/task/1377?kanban-status=1224894

## How to test?

Please do only code review.
This PR is related to:
* https://github.com/rero/rero-ils-ui/pull/208
* https://github.com/rero/ng-core/pull/151

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -71,7 +71,7 @@ if [ $# -eq 0 ]
 
         set -e
         # remove this when https://github.com/pypa/pipenv/issues/4147 will be solved
-        pipenv check -i 37752
+        pipenv check -i 37752 -i 38076
```
Comment (rerowep) [inline run-tests.sh]: We dont need `-i 38076` it was solved with https://github.com/rero/rero-ils/pull/854

File: rero_ils/modules/documents/serializers.py [hunk 2]
```diff
@@ -52,51 +52,64 @@ def post_process_serialize_search(self, results, pid_fetcher):
...
-                metadata['available'] = available
-
+            if viewcode != current_app.config.get(
+                    'RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'
+            ):
+                # Compute availibility using viewcode
+                # and filter items by viewcode
+                viewcode_organisation_pid = \
+                    Organisation.get_record_by_viewcode(viewcode)['pid']
+                for record in records:
```
Comment (rerowep) [inline rero_ils/modules/documents/serializers.py]: Maybe using of https://www.programiz.com/python-programming/methods/built-in/filter makes this code clearer.

File: rero_ils/query.py [hunk 3]
```diff
@@ -37,23 +37,26 @@ def document_search_factory(self, search, query_parser=None):
     agg_size = current_app.config.get('RERO_ILS_AGGREGATION_SIZE').get(
         'documents', current_app.config.get(
             'RERO_ILS_DEFAULT_AGGREGATION_SIZE'))
-    view = request.args.get(
-        'view', current_app.config.get('RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'))
-    if view == current_app.config.get('RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'):
+    view = request.args.get('view')
+    if view is None:
+        return view_search_factory(self, search, query_parser)
+    if view == current_app.config.get(
```
Comment (rerowep) [inline rero_ils/query.py]: Why splitting the line if it has less 80 characters ?

File: rero_ils/query.py [hunk 4]
```diff
@@ -37,23 +37,26 @@ def document_search_factory(self, search, query_parser=None):
     agg_size = current_app.config.get('RERO_ILS_AGGREGATION_SIZE').get(
         'documents', current_app.config.get(
             'RERO_ILS_DEFAULT_AGGREGATION_SIZE'))
-    view = request.args.get(
-        'view', current_app.config.get('RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'))
-    if view == current_app.config.get('RERO_ILS_SEARCH_GLOBAL_VIEW_CODE'):
+    view = request.args.get('view')
+    if view is None:
```
Comment (rerowep) [inline rero_ils/query.py]: Do we have tree cases here?
1. view is None
2. view == current_app.config(...
3. else

Comment (AoNoOokami) [reply rero_ils/query.py]: Yes, are you proposing some improvement?

Comment (rerowep) [reply rero_ils/query.py]: There is no switch in python only:
if .....
elif .....
elif .....

**Comment by AoNoOokami** [comment]
>     1. Doing a search from the main bar doesn't filter by default to my organisation's documents, but it should.
> 
>     2. The organisation facet does not include/count ebooks, but it should.
>        image
> 
>     3. When opening a 1st-level value in the facet, all the 2nd-level values are displayed. Is it possible to display the 10 first in count, with a button to expand (as for the authors)?
>        image
> 
>     4. In the public global view, the facet "Organisation" could be renamed into "Library" (more user friendly).

Ping @jma and @sebastiendeleze

**Comment by AoNoOokami** [comment]
@pronguen : The correction of points 1 to 3 are in this PR: https://github.com/rero/ng-core/pull/174. As soon as it's approved and released, the corresponding PRs will be deployed again for testing.

**Comment by AoNoOokami** [comment]
Remarks 1 to 3 were corrected, deployed and tested. As several people from PO group validated the corrections (see https://gitter.im/rero/reroils-PO?at=5ea2f67fe5ed621d4dda278e and following messages), I merge this PR.

**Pull request merged**